### DISCUSSION TITLE
Customize ObjectVar Display

### DISCUSSION THREAD

**Discussion by netixx** [netbox-community/netbox#6018] [Q&A] [answered]
Hello,

I have a script where I need to select a bunch of interfaces, I am using this snippet:
```
lag_chooser = MultiObjectVar(
        label="Interfaces",
        description="Choose Interfaces to deploy vlan on",
        required=True,
        model=Interface,
        query_params={
            'type': InterfaceTypeChoices.TYPE_LAG,
        },
    )
```
But it only displays the "name" of the interface, but I need to display the device as well.

Is there any way to do this in scripts or does it require a plugin ?

**Comment by netixx** [comment]
My investigation pointed me to netbox/netbox/project-static/js/forms.js line 200.

This is the line that displays the name of the record `record.text = record[element.getAttribute('display-field')] || record.name;`.
I see two solutions if we want to keep the dynamic loading via API:
- edit the javascript (but I don't know how to overload it in a script)
- create an API endpoint that sends a view of the Interface, that overrides/adds a field containing the informations I want to display
 
I am also exploring ways to make the field static by using "StaticSelect2Multiple", but no luck so far...

**Comment by jeremystretch** [comment, marked as answer]
No, this is not supported.

**Comment by netixx** [comment]
I found a workaround using a MultiChoiceVar

```
lag_chooser = MultiChoiceVar(
        label="Interfaces",
        description="Choose Interfaces to deploy vlan on",
        required=True,
        widget=StaticSelect2Multiple(),
        choices=(
            (interface.id, "{} {}".format(interface.device.name, interface.name))
            for interface in Interface.objects.filter(
                type=InterfaceTypeChoices.TYPE_LAG,
            )
        )
    )
```

instead of 
```
lag_chooser = MultiObjectVar(
         label="Interfaces",
         description="Choose Interfaces to deploy vlan on",
         required=True,
         model=Interface,
         query_params={
             'type': InterfaceTypeChoices.TYPE_LAG,
         },
     )
```

**Reply by jeremystretch** [reply]
Note that this approach will force population of _all_ available values at render time, rather than using the REST API backend. This can lead to significant delays in page loading time when there are thousands of choices available.

**Reply by netixx** [reply]
yes, I understand, however, is there another way to achieve this ?

Maybe through a plugin with a custom component/field ?
